### PR TITLE
separate export map from its builder

### PR DESCRIPTION
**Pull request by soryy708**
`ExportMap` is currently a very large file that is hard to follow, and therefore hard to maintain.

I identified that its static methods, which are responsible for building instances of `ExportMap`, can be generalized to the [builder software design pattern](https://en.wikipedia.org/wiki/Builder_pattern).

Lets make the separation, to make it clear that instances of `ExportMap` aren't coupled to any of the globals or procedures that are currently in `ExportMap.js`.

It's also good to rename the files and classes appropriately.

### CODE REVIEW

File: tests/src/core/getExports.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import sinon from 'sinon';
 import eslintPkg from 'eslint/package.json';
 import typescriptPkg from 'typescript/package.json';
 import * as tsConfigLoader from 'tsconfig-paths/lib/tsconfig-loader';
-import ExportMap from '../../../src/ExportMap';
+import ExportMap from '../../../src/exportMapBuilder';
```
Comment (ljharb) [inline tests/src/core/getExports.js:7]:
```suggestion
import Exports from '../../../src/exportMapBuilder';
```
should this be named "Exports" or "ExportMapBuilder"?

**Review by ljharb — APPROVED** [review]
Assuming that the ExportMap class is moved without changes, this LGTM!

File: tests/src/core/getExports.js (hunk 1 above)
Comment (soryy708) [reply tests/src/core/getExports.js]: Fair.

- https://github.com/import-js/eslint-plugin-import/pull/2985/commits/cc1b19e6417c5f28ba2ade6db6477a24a8f42b3b

**Review by ljharb — APPROVED** [review]
